agent/lib/kontena/service_pods/creator.rb

```diff
@@ -78,21 +80,41 @@ def perform
         service_container
       end
 
-      # @param [Docker::Container] service_container
-      # @param [String] type
       # @return [Boolean]
-      def run_hooks(service_container, type)
-        service_pod.hooks.each do |hook|
-          if hook['type'] == type
-            info "running #{type} hook: #{hook['cmd']}"
-            command = ['/bin/sh', '-c', hook['cmd']]
-            log_hook_output(service_container.id, ["running #{type} hook: #{hook['cmd']}"], 'stdout')
-            stdout, stderr, exit_code = service_container.exec(command)
-            log_hook_output(service_container.id, stdout, 'stdout')
-            log_hook_output(service_container.id, stderr, 'stderr')
-            if exit_code != 0
-              raise "Failed to execute hook: #{hook['cmd']}"
+      def run_pre_start_hooks
+        hooks_for('pre_start').each do |hook|
+          service_config = config_container(service_pod.dup)
+          service_config['Cmd'] = ['/bin/sh', '-c', hook['cmd']]
+          service_container = nil
+          begin
+            service_container = create_container(service_config)
```

Comment:
Using the complete `service_pod.service_config` including all labels etc. means that running the pre-start hooks will also have some side-effects, like `kontena.local` DNS and LB registrations? That means that client traffic might start getting routed to the pre-start container, which will likely fail?